eureka/S3_data_reduction/bright2flux.py

```diff
@@ -183,19 +184,30 @@ def convert_to_e(data, meta, log):
     meta:   MetaClass
         The metadata object.
     """
+    if data.shdr['BUNIT'] != 'ELECTRONS/S':
+        log.writelog('Automatically getting reference files to convert units to e/s')
+        if data.mhdr['TELESCOP'] != 'JWST':
+            raise ValueError('Currently unable to automatically download reference files for non-jwst observations!')
+        meta.photfile, meta.gainfile = download_ancil(data.filename)
+    
     if data.shdr['BUNIT'] == 'MJy/sr':
         # Convert from brightness units (MJy/sr) to flux units (uJy/pix)
         # log.writelog('Converting from brightness to flux units')
         # data = b2f.bright2flux(data, data.shdr['PIXAR_A2'])
         # Convert from brightness units (MJy/sr) to DNs
-        log.writelog('  Converting from brightness units (MJy/sr) to electrons')
-        meta.photfile = os.path.join(meta.topdir, *meta.ancildir.split(os.sep)) + '/' + data.mhdr['R_PHOTOM'][7:]
+        log.writelog('  Converting from brightness units (MJy/sr) to electrons per second (e/s)')
         data = bright2dn(data, meta)
-        meta.gainfile = os.path.join(meta.topdir, *meta.ancildir.split(os.sep)) + '/' + data.mhdr['R_GAIN'][7:]
         data = dn2electrons(data, meta)
     if data.shdr['BUNIT'] == 'DN/s':
         # Convert from DN/s to e/s
         log.writelog('  Converting from data numbers per second (DN/s) to electrons per second (e/s)')
-        meta.gainfile = os.path.join(meta.topdir, *meta.ancildir.split(os.sep)) + '/' + data.mhdr['R_GAIN'][7:]
         data = dn2electrons(data, meta)
+    
     return data, meta
+
+def download_ancil(fitsname):
```

Comment:
We should not require internet access for users to be able to reduce data, so these downloaded files need to go somewhere (like an ancil directory) that can be checked later.

Reply:
The code uses the CRDS variables set in your environment (.bashrc file or equivalent), and it always checks to see if you have the required files first before trying to download anything from the internet. If you have the required files, no internet connection required. The files get saved wherever your CRDS cache is setup to be, which saves you the hassle of redownloading files if you ran stage 2 which would've downloaded the files already. The slight inconvenience is that it checks that you have the right gain and photom files for every file you reduce, but that could also be a feature if you have multiple visits or instruments in the same folder.

Reply:
Got it.  There's a lot going on under the hood here that isn't obvious.  Consider adding more comments and renaming `download_ancil` to something like `retrieve_ancil` so that there's no confusion.